ui/Dockerfile

```diff
@@ -0,0 +1,23 @@
+# syntax=docker.io/docker/dockerfile:1.7-labs
+FROM node:18
+
+ARG PROXY_HOST=backend
+WORKDIR /ui
+
+COPY package.json yarn.lock ./
+
+# TODO: configure this directly from the ui app
+# Work around to set the proxy host; required for running locally with docker-compose
+RUN <<EOF
+set -ex
+apt-get update
+apt-get install -y jq
+jq --arg proxy "http://$PROXY_HOST:8080" '.proxy = $proxy' package.json > temp.json
```

Comment:
Hey 👋 @jamieknight-db! I am trying to parameterize the UI react app to use the api url `http://backend:8080` instead of `http:localhost:8080` when running inside of docker compose. My react knowledge is pretty dismal; is there a better way to configure the react app than just replacing the `proxy` key inside `package.json`?

Reply:
For context, this PR enables us to spin up the frontend and backend with just `docker compose up -d`

Reply:
@tnightengale to be honest, I am unsure the best way to parameterize this. Ideally, all parameterized variables are stored in the `.env` file and could be accessed from the package.json scripts. 

```
   "proxy": "${REACT_APP_API_URL}"
```

Or we might need to utilize a setupProxy.js file and createProxyMiddleware and go that route? 

Maybe @bijanhoule or @creechy or @yc-shawn can chime in?